### ISSUE TITLE
Extracting text-lines

### ISSUE THREAD

**Issue by MrOCR** [mittagessen/seg#4]
If I understand correctly, `seg pred` will predict the baseline labels.
After that, how can we extract text-lines from the original images? @mittagessen

**Comment by MrOCR** [comment]
@mittagessen [pseg2lines](https://github.com/NVlabs/ocroseg/blob/master/pseg2lines)

**Comment by mittagessen** [comment]
ocroseg doesn't converge for me either, even when using the UW3 dataset the model provided with it was supposedly trained on. During training it starts to erode lines and output usually goes completely blank after ~70 epochs.

**Comment by MrOCR** [comment]
@mittagessen use the [**`entire uw3 framed-lines dataset 264MB,`**](http://storage.googleapis.com/lpr-ocr/uw3-framed-lines.tgz) also use the [**`bigmodel.py`**](https://github.com/NVlabs/ocroseg/blob/master/bigmodel.py) to create a more complex model.

[This link contains all the datasets!](http://storage.googleapis.com/lpr-ocr/) 
Note that: If you want to create a resilient model, use the high-skew/uw3-framed-lines-degraded-**

@mittagessen please copy the links fast, because I will delete them before others see this!!

**Comment by mittagessen** [comment]
The issue isn't the size of the training set (I used a subset of UW with ~1450 images and have plenty of other training data that doesn't work either) as I can't even get the net to overfit on a single sample no matter if using weighted gradient (ocroseg) or not (one version of my code).

**Comment by MrOCR** [comment]
@mittagessen I do understand.
Have a look at [**`this repo,`**](https://github.com/watersink/ocrsegment) the developer had moded the ocroseg code, along with porting it to python3.

**Comment by MrOCR** [comment]
Also have a look at https://github.com/lquirosd/P2PaLA

**Comment by MrOCR** [comment]
@mittagessen 
- Please post the `entire command` that you used to train ocroseg, along with the `model network structure` script, so that I could test.
- Also, what accuracy & loss are you getting.

**Comment by MrOCR** [comment]
https://github.com/dhlab-epfl/dhSegment
https://github.com/lquirosd/P2PaLA

**Comment by MrOCR** [comment]
![11](https://user-images.githubusercontent.com/33478216/46333761-3e3e5a80-c62a-11e8-8ea5-a00037f290ef.gif)

Here is a list of things I tested:
[Ocroseg](https://github.com/NVlabs/ocroseg): The process always failed and stopped automatically.
[ocrsegment](https://github.com/watersink/ocrsegment): It didn't fit, output is blank.
[pix2pix](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix): Required down-sizing of images, resource intensive training, then up-sizing of images. Too messy.
[pix2pixHD](https://github.com/NVIDIA/pix2pixHD): I didn't test

@mittagessen good god, it's a nightmare trying to train a functional base-line detection model!
Why does every method I tested failed?
How does tmbdev gets ocroseg to work? is it because of the network structure?

The only method that seems to get good results is based on scene text detection, example:
[ text-detection-ctpn](https://github.com/eragonruan/text-detection-ctpn)

**Comment by mittagessen** [comment]
A simple fully convolutional baseline detector does work for baseline detection and the current code in the `baseline` branch does it fairly well. The process seems to be a bit finicky with the learning rate and only converges reliably with SGD.

I've been busy with other things the last few days so I haven't been able to get the hyperparameters for the new convex hull target expansion correct but if you don't mind somewhat noisy pixel maps that one works also as designed if you comment out the `convex_hull_image` line in `dataset.py`.

**Comment by MrOCR** [comment]
@mittagessen is dhSegment worth testing?

**Comment by MrOCR** [comment]
https://github.com/PRHLT/BaseLinePage

**Comment by MrOCR** [comment]
Text Line Segmentation for Challenging Handwritten Document Images using Fully Convolutional Network
https://github.com/beratkurar/textline-segmentation-using-fcn
https://www.cs.bgu.ac.il/~berat/papers/Icfhr2018Text.pdf
https://www.cs.bgu.ac.il/~berat/papers/Icfhr2018TextPoster.pdf
